packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1090,6 +1094,216 @@ describe("SharedTree", () => {
 				});
 			}
 		});
+
+		describe("can resubmit", () => {
+			const sb = new SchemaBuilder({ scope: "shared tree undo tests" });
+			const schema = sb.intoSchema(sb.list(sb.list(sb.string)));
+
+			/**
+			 * Each test will involve two peers, one of which has removed a tree (and is therefore able to restore it) -
+			 * let's call that one peer 1.
+			 *
+			 * Peer 1 can do one of three things:
+			 * (A) Commit 1: make an edit to the removed tree, Commit 2: make another edit to removed tree
+			 * (B) Commit 1: make an edit to the removed tree, Commit 2: restore the removed tree
+			 * (C) Commit 1: restore the removed tree, Commit 2: make an edit to the restored tree
+			 * Peer 2 can only do option (A) above.
+			 */
+			const scenarios = [
+				["A", "A"],
+				["A", "B"],
+				["A", "C"],
+				["B", "A"],
+				["C", "A"],
+			];
+
+			function makeTestTitle(peer: string): string {
+				switch (peer) {
+					case "A": {
+						return "an edit to the removed tree after another edit";
```

Comment:
This "X after Y" phrasing makes it sound as though only X is being resubmitted. How about:
A => "two edits to the removed tree"
B => "the restoration and editing of the removed tree"
C => "the editing and restoration of the removed tree"

Reply:
From the task description (emphasis added):
...the commit*s* sequenced last go through a resubmit phase such that *they* rebase over the commits sequenced first
...the "x 2" factor comes from controlling which of peer1's commit*s* or peer2's commits get sequenced later (and go through the resubmit code path).